aws/resource_aws_s3_bucket_object.go

```diff
@@ -527,8 +527,27 @@ func validateMetadataIsLowerCase(v interface{}, k string) (ws []string, errors [
 }
 
 func resourceAwsS3BucketObjectCustomizeDiff(_ context.Context, d *schema.ResourceDiff, meta interface{}) error {
-	if d.HasChange("etag") {
-		d.SetNewComputed("version_id")
+
+	for _, key := range []string{
+		"cache_control",
+		"content_base64",
+		"content_disposition",
+		"content_encoding",
+		"content_language",
+		"content_type",
+		"content",
+		"etag",
+		"kms_key_id",
+		"metadata",
+		"server_side_encryption",
+		"source",
+		"storage_class",
+		"website_redirect",
+	} {
+		if d.HasChange(key) {
+			return d.SetNewComputed("version_id")
+		}
+
 	}
```

Comment:
We have the function call `d.HasChanges()` that can take multiple keys

```suggestion
	if d.HasChanges("cache_control", "content_base64", "content_disposition", "content_encoding", "content_language", "content_type", "content", "etag", "kms_key_id", "metadata", "server_side_encryption", "source", "storage_class", "website_redirect") {
		return d.SetNewComputed("version_id")
	}
```

Reply:
@gdavison unfortunately `d.HasChanges` is not in `d.ResourceDiff`. I did refactor another instance in this file where we can use `d.hasChanges`.